lib/chef/resource/build_essential.rb

```diff
@@ -151,6 +152,20 @@ def xcode_cli_installed?
           # pkgutil returns an error if the package isn't found aka not installed
           cmd.error? ? false : true
         end
+
+        #
+        # Return to package label of the latest XCode Command Line Tools update, if available
+        #
+        # @return [String]
+        def xcode_cli_package
+          cmd = <<-EOH.gsub(/^ {14}/, "")
+          softwareupdate -l | grep "\*.*Command Line" | tail -n 1 | awk -F"*" '{print $2}' | sed -e 's/^ *//' | tr -d '\n' | sed 's/Label: //g'
```

Comment:
Instead of piping this to various unix utilties we should parse it in Ruby directly. Can you post the two possible text strings. I can help get this done natively in Ruby, which will be a bit faster and less brittle.

Reply:
I started trying to use `String` utils to achieve this, but rapidly started regexing. So here's my attempt (see latest commit for full parse): 
```ruby

xcode_regex = /\* (Label: )?(?<label>Command Line Tools for Xcode-(?<version>\d{2,}\.\d{1,2}))/`

```

Example output from `man softwareupdate` on macOS 10.15.2

```
             * Label: MacBookAirEFIUpdate2.4-2.4
                   Title: MacBook Air EFI Firmware Update, Version: 2.4, Size: 3817K, Recommended: YES, Action: restart,
              * Label: ProAppsQTCodecs-1.0
                   Title: ProApps QuickTime codecs, Version: 1.0, Size: 968K, Recommended: YES,
              * Label: JavaForOSX-1.0
                   Title: Java for OS X 2012-005, Version: 1.0, Size: 65288K, Recommended: YES,
```

Same from macOS 10.14.4:

```
             * MacBookAirEFIUpdate2.4-2.4
                   Title: MacBook Air EFI Firmware Update, Version: 2.4, Size: 3817K, Recommended: YES, Action: restart,
              * ProAppsQTCodecs-1.0
                   Title: ProApps QuickTime codecs, Version: 1.0, Size: 968K, Recommended: YES,
              * JavaForOSX-1.0
                   Title: Java for OS X 2012-005, Version: 1.0, Size: 65288K, Recommended: YES,
```

The exact strings we want to match on are:

`< 10.15`

```
Software Update Tool

Finding available software
Software Update found the following new or updated software:
* Command Line Tools for Xcode-11.2
	Command Line Tools for Xcode, Version: 11.2, Size: 224942K, Recommended: YES,
* Command Line Tools for Xcode-11.3
	Title: Command Line Tools for Xcode, Version: 11.3, Size: 224878K, Recommended:
```

`>= 10.15`

```
Software Update Tool

Finding available software
Software Update found the following new or updated software:
* Label: Command Line Tools for Xcode-11.2
	Title: Command Line Tools for Xcode, Version: 11.2, Size: 224942K, Recommended: YES,
* Label: Command Line Tools for Xcode-11.3
	Title: Command Line Tools for Xcode, Version: 11.3, Size: 224878K, Recommended:
```